### ISSUE TITLE
L028 Rule fails with declared variables in TSQL dialect

### ISSUE THREAD

**Issue by timvink** [sqlfluff/sqlfluff#1868]
## Expected Behaviour

This should not give any linting errors:

```
DECLARE @id_date int;
SET @id_date = 20211108;

SELECT sometable.some_column
FROM sometable
WHERE 1 = 1
    AND sometable.column_with_date = @id_date
```

## Observed Behaviour

```
= [query.sql] FAIL
L:   7 | P:  38 | L028 | Unqualified reference '@id_date' found in single table
                       | select which is inconsistent with previous references.
```

## Steps to Reproduce

```shell
mkdir test_sql
cd test_sql
echo "[sqlfluff]\ndialect = tsql" > .sqlfluff

echo "DECLARE @id_date int;\n" > query.sql
echo "SET @id_date = 20211108;\n\n" >> query.sql
"SELECT sometable.some_column\n" >> query.sql
"FROM sometable\n" >> query.sql
"WHERE 1 = 1\n" >> query.sql
"    AND sometable.column_with_date = @id_date\n" >> query.sql

pip install sqlfluff==0.8.1
sqlfluff lint query.sql
```

## Dialect

`tsql`

## Version

`Python 3.9.5`
`sqlfluff, version 0.8.1`

## Configuration

```
# .sqlfluff
[sqlfluff]
dialect = tsql
```

**Comment by timvink** [comment]
Actually, the same situation occurs with rule `L027` as well

**Comment by timvink** [comment]
Thanks 👍
